src/QsCompiler/SyntaxProcessor/SymbolOccurrence.fs

```diff
@@ -0,0 +1,194 @@
+        | ControlledApplication e -> inExpression e
+        | ArrayItem (e1, e2)
+        | ADD (e1, e2)
+        | SUB (e1, e2)
+        | MUL (e1, e2)
+        | DIV (e1, e2)
+        | MOD (e1, e2)
+        | POW (e1, e2)
+        | EQ (e1, e2)
+        | NEQ (e1, e2)
+        | LT (e1, e2)
+        | LTE (e1, e2)
+        | GT (e1, e2)
+        | GTE (e1, e2)
+        | AND (e1, e2)
+        | OR (e1, e2)
+        | BOR (e1, e2)
+        | BAND (e1, e2)
+        | BXOR (e1, e2)
+        | LSHIFT (e1, e2)
+        | RSHIFT (e1, e2)
+        | SizedArray (e1, e2)
+        | RangeLiteral (e1, e2)
+        | CallLikeExpression (e1, e2) -> inExpression e1 @ inExpression e2
+        | CONDITIONAL (e1, e2, e3)
+        | CopyAndUpdate (e1, e2, e3) -> inExpression e1 @ inExpression e2 @ inExpression e3
+        | ValueTuple es
+        | StringLiteral (_, es)
+        | ValueArray es -> Seq.collect inExpression es |> Seq.toList
+        | NewArray (t, e) -> inType t @ inExpression e
+        | InvalidExpr -> []
+
+    let rec inInitializer initializer =
+        match initializer.Initializer with
+        | QubitRegisterAllocation e -> inExpression e
+        | QubitTupleAllocation xs -> Seq.collect inInitializer xs |> Seq.toList
+        | SingleQubitAllocation
+        | InvalidInitializer -> []
+
+    let inGenerator generator =
+        match generator.Generator with
+        | Intrinsic
+        | AutoGenerated
+        | FunctorGenerationDirective _ -> []
+        | UserDefinedImplementation s -> symbolDeclarations s
+
+    let rec inNamedTuple =
+        function
+        | QsTuple xs -> Seq.collect inNamedTuple xs |> Seq.toList
+        | QsTupleItem (s, t) -> symbolDeclarations s @ inType t
+
+    let inCallable (callable: CallableDeclaration) =
+        let typeParams = Seq.map Declaration callable.Signature.TypeParameters |> Seq.toList
+        let args = inNamedTuple callable.Signature.Argument
+        let returnType = inType callable.Signature.ReturnType
+        Declaration callable.Name :: typeParams @ args @ returnType
+
+    [<CompiledName "InFragment">]
+    let inFragment fragment =
+        match fragment with
+        | NamespaceDeclaration s -> [ Declaration s ]
+        | OpenDirective (_, s) -> QsNullable<_>.Choose id [ s ] |> Seq.toList |> List.map Declaration
+        | DeclarationAttribute (s, e) -> UsedVariable s :: inExpression e
+        | OperationDeclaration c
+        | FunctionDeclaration c -> inCallable c
+        | BodyDeclaration g
+        | AdjointDeclaration g
+        | ControlledDeclaration g
+        | ControlledAdjointDeclaration g -> inGenerator g
+        | ImmutableBinding (s, e)
+        | MutableBinding (s, e)
+        | ForLoopIntro (s, e) -> symbolDeclarations s @ inExpression e
+        | UsingBlockIntro (s, i)
+        | BorrowingBlockIntro (s, i) -> symbolDeclarations s @ inInitializer i
+        | TypeDefinition t -> Declaration t.Name :: inNamedTuple t.UnderlyingType
```

Comment:
This might run into trouble when items are unnamed; before Invalid and Missing symbols were not collected, now they are.

Reply:
From testing, I couldn't find any issues with unnamed items in type declarations. Note that the omitted symbol in this case is missing a range, so the language server should ignore it.